dhall/src/Dhall/TypeCheck.hs

```diff
@@ -52,6 +52,8 @@ import qualified Dhall.Set
 import qualified Dhall.Pretty.Internal
 import qualified Dhall.Util
 
+type X = Void
```

Comment:
A bit of documentation for why we have this type alias would be good.

Also: Do we want to keep it forever, or are we going to deprecate it?

Reply:
@sjakobi: We'll eventually deprecate it